src/util/rsthreads.cc

```diff
@@ -249,8 +264,8 @@ void RsQueueThread::threadTick()
     }
     else
     {
-        uint32_t deltaT = now - mLastWork;
-        double frac = deltaT / mRelaxFactor;
+        uint64_t deltaT = now - mLastWork;
```

Comment:
at midnight, this is going to cause an int overflow: mLastWork=86399 (last second of previous day) and now = 0 => deltaT = approximately 10^31

Reply:
actually no, because 1000*getCurrentTS() is the number of ms since 1/1/70, it never returns to 0

Reply:
Ah you're right! So getCurrentTS() is a double, which means it has 52 bits of precision. timebuf.time is 64 bits but effectively uses 32, and adding milliseconds you need approximately 42 bits. That gives us some slack. Making getCurrentTS_ms() returning miliseconds as a uint64_t would probably avoid an implicit cast (line 255) and give you more space, but it's not critical here.

Reply:
I use getCurrentTS() written by DrBob since long. It very well suits my need everytime I need to do calculations. It is a pretty useful method that should  be available everywhee in the code without the need to redefine it.